qualtran/bloqs/qft/qft_text_book.py

```diff
@@ -40,7 +40,7 @@ class QFTTextBook(GateWithRegisters):
 
     @cached_property
     def signature(self) -> 'Signature':
-        return Signature.build(q=self.bitsize)
+        return Signature([Register("q", QBit() if self.bitsize == 1 else QUInt(self.bitsize))])
```

Comment:
I think we should not special case on self.bitsize == 1; do we have a consensus yet?

Reply:
sorry I prematurely pushed, I'm replacing with build_from_dtypes which will handle this.

Reply:
Or rather we aren't special casing for QUInt / QInt (1)